### PR TITLE
ui: refactor model updating and rendering into ModelRenderer class

### PR DESCRIPTION
**Pull request by deanlee**
This PR preserves most of the original logic from before the refactor. The key change is replacing the previous method of validating messages, which relied on `sm.rcv_frame("modelV2") > uiState()->scene.started_frame`, with a new check using `sm.alive("modelV2")`.  consistent with the approach used in other modules like `hudRenderer`.

### CODE REVIEW

**Comment by maxime-desroches** [comment]
trigger-jenkins

**Comment by maxime-desroches** [comment]
trigger-jenkins

**Comment by maxime-desroches** [comment]
Nice, much easier to review

File: selfdrive/ui/qt/onroad/model.cc [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+void ModelRenderer::draw(QPainter &painter, const QRect &surface_rect) {
+  auto &sm = *(uiState()->sm);
+  if (sm.updated("carParams")) {
+    longitudinal_control = sm["carParams"].getCarParams().getOpenpilotLongitudinalControl();
+  }
+
+  // Check if data is up-to-date
+  if (!(sm.alive("liveCalibration") && sm.alive("modelV2"))) {
```
Comment (maxime-desroches) [inline selfdrive/ui/qt/onroad/model.cc:23]: Is that logic really equivalent to the old one? https://github.com/commaai/openpilot/blob/7fa8e3d8a65682a00d8ee6d5dba2b58292921dfc/selfdrive/ui/ui.cc#L151-L154

Comment (deanlee) [reply selfdrive/ui/qt/onroad/model.cc:23]: It's not exactly the same, but it accomplishes the same functionality, and perhaps even better.
First, `ModelRenderer::draw()` is only called when onroad, so there’s no need to check if `scene.started` is true within `draw()`.
Second, `sm.alive(xxx)` is similar to `sm.rcv_frame("xxx") > scene.started_frame` and has the added benefit of skipping the rendering of invalid model data on the screen if the messages are already inactive for some reason. (At this point, the OnroadAlerts class may have already started displaying warnings.)

**Pull request merged**